### PR TITLE
depricate DnsProxyNotRequiredForNetworkRule switch parameter for New-…

### PR DESCRIPTION
**Pull request by b564518128**
## Description
Depricate `-DnsProxyNotRequiredForNetworkRule` switch parameter for `New-AzFirewall`
Depricate `-ProxyNotRequiredForNetworkRule` switch parameter for `New-AzFirewallPolicyDnsSetting`

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Network/Network/ChangeLog.md [hunk 1]
```diff
@@ -33,6 +33,11 @@
 * Modified the warning message for `New-AzLoadBalancerFrontendIpConfig`, `New-AzPublicIpAddress` and `New-AzPublicIpPrefix`.
 * Added VpnGatewayIpConfigurations to `Get-AzVpnGateway` output
 * Bug fix for Set-AzApplicationGatewaySslCertificate (https://github.com/Azure/azure-powershell/issues/9488)
+* [Breaking Change] Depricated a switch parameter in below cmdlets
```
Comment (isra-fel) [inline src/Network/Network/ChangeLog.md:36]: nit: typos in below change log section
```suggestion
* [Breaking Change] Deprecated a switch parameter in below cmdlets
```

**Review by isra-fel — CHANGES_REQUESTED** [review]
Hi JieJiong, are the parameters already deprecated on the sevice side or will be deprecated in the future? And when do you plan to remove them from cmdlets?

File: src/Network/Network/Models/AzureFirewallPolicy/PSAzureFirewallPolicyDnsSetting.cs [hunk 2]
```diff
@@ -22,7 +22,5 @@ public class PSAzureFirewallPolicyDnsSettings
         public IList<string> Servers { get; set; }
 
         public bool? EnableProxy { get; set; }
-
-        public bool? RequireProxyForNetworkRules { get; set; }
```
Comment (isra-fel) [inline src/Network/Network/Models/AzureFirewallPolicy/PSAzureFirewallPolicyDnsSetting.cs:26]: Besides, we usually announce the breaking change at least 2 sprints prior to the major release, in order not to surprise our customers.

Comment (iatodoro) [reply src/Network/Network/Models/AzureFirewallPolicy/PSAzureFirewallPolicyDnsSetting.cs:26]: This particular argument is related to a feature currently in Preview and we're taking the opportunity to streamline the user experience prior to GA, which is coming up in the beginning of November. 

The service will continue to accept the parameter if provided, but its actual functionality will stop working in the next week or two.

**Comment by VeryEarly** [comment]
Hi @iatodoro ,

this network-july branch won't be able to catch 2020-10-27 powershell release, since your PR contains breaking changes, please pick your commits and make a new PR to master branch.

close this PR for now.

**Comment by iatodoro** [comment]
Why is this closed now? We submitted the change nearly 2 weeks ago in order to make the 10-27 release -we did not expect so much pushback on changing a preview feature. Further, if this is NOT merged, it WILL become a breaking change as customer will expect this parameter to work, but it will not, as the backend implementation is changing (not negotiable).

Please reconsider this as it will lead to a poorer experience in the long run.

**Comment by VeryEarly** [comment]
> Why is this closed now? We submitted the change nearly 2 weeks ago in order to make the 10-27 release -we did not expect so much pushback on changing a preview feature. Further, if this is NOT merged, it WILL become a breaking change as customer will expect this parameter to work, but it will not, as the backend implementation is changing (not negotiable).
> 
> Please reconsider this as it will lead to a poorer experience in the long run.

Please check my comment "please pick your commits and make a new PR to master branch", this is how this PR is going to go with this release.

Network-July release branch was delayed since API 2020-07 was not ready. All RP that need to released at 2020-10-27 need to be merged into master branch now.

I checked your PR before closed, CI was not passed yet and there are comments unresolved, so not ready to be merged yet.

Please cherry pick these 3 commits into a new branch (created from master) and create a new PR against master branch. I will be reviewed and merged into master branch for release 2020-10-27.

Thanks.

**Comment by iatodoro** [comment]
@VeryEarly thanks for clarifying! 

@b564518128 let's follow up on this quickly to make sure it merges for release 10-27.

**Pull request closed without merging**